### PR TITLE
Add/gx video

### PR DESCRIPTION
**Pull request by MarkMed**
New gx-video component to display videos _(from Youtube for now)_

### CODE REVIEW

File: src/components/video/video.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     *
+    @Prop() lazyLoad = true;
+  */
+
+  /**
+   * Emitted when the element is clicked.
+   */
+  @Event() onClick: EventEmitter;
+
+  handleClick(event: UIEvent) {
```
[the remark this answers is not in the document]
Comment (MarkMed) [reply src/components/video/video.tsx]: Fix this in the commit 7a2a06f 👌

File: src/components/video/video.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
+@import "../common/_base";
+
+gx-video {
+  /**
+    * @prop --gx-video-height:
+    * Set the height or the control.
+    * (500px by default)
+    */
+  --gx-video-height: 500px;
```
[the remark this answers is not in the document]
Comment (MarkMed) [reply src/components/video/video.scss]: I do not know if the sizes will be assigned in the control itself or inside in the control _(in the video container or in the iframe)_
Anyway, I left this feature but with `100%` of size for the video container _(iframe container)_ 

Commit c5ec89c

File: src/components/video/video.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  private parseYoutubeSrc(src) {
+    const domain_ID_Array = src.split("watch?v=");
+    return `${domain_ID_Array[0]}embed/${domain_ID_Array[1]}`;
+  }
+
+  render() {
+    return (
+      <div class="gxVideoContainer">
+        <iframe src={this.parseYoutubeSrc(this.src)} />
```
[the remark this answers is not in the document]
Comment (MarkMed) [reply src/components/video/video.tsx]: I leave the comment in the PR message.
I just added the explanation in the doc of the control with an extra usage sample.
See in 01919a5

File: src/components/video/video.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+  /**
+   * This attribute lets you specify how this element will behave when hidden.
+   *
+   * | Value        | Details                                                                     |
+   * | ------------ | --------------------------------------------------------------------------- |
+   * | `keep-space` | The element remains in the document flow, and it does occupy space.         |
+   * | `collapse`   | The element is removed form the document flow, and it doesn't occupy space. |
+   */
+  @Prop() invisibleMode: "collapse" | "keep-space" = "collapse";
```
[the remark this answers is not in the document]
Comment (MarkMed) [reply src/components/video/video.tsx]: implemented in 4028fb1 👌

File: src/components/video/readme.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# gx-video
+
+A component to load a video.
+
+## Video Support
+
+By now, only supports youtube videos. Future updates will support videos URLs from any platform.
```
Comment (ncardeli) [inline src/components/video/readme.md]:
```suggestion
At the moment it only supports YouTube videos. Future updates will support videos URLs from other video publishers and single video files.
```

File: src/components/video/readme.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+## Usage Sample
+
+```HTML
+<gx-video src="https://www.youtube.com/watch?v=k2jT6TlxEOM" style="--gx-video-height: 50vh;"></gx-video>
```
Comment (ncardeli) [inline src/components/video/readme.md]: 👍

File: src/components/video/video.scss [hunk 7]
```diff
@@ -0,0 +1,23 @@
+@import "../common/_base";
+
+gx-video {
+  /**
+    * @prop --gx-video-height:
+    * Set the height of the control from the video container or the control itself.
+    * (100% by default)
+    */
+  --gx-video-height: 100%;
```
Comment (ncardeli) [inline src/components/video/video.scss]: Let's add another one for the width.

File: src/components/video/readme.md (hunk 6 above)
Comment (MarkMed) [reply src/components/video/readme.md]: Good?

File: src/components/video/readme.md (hunk 5 above)
Comment (MarkMed) [reply src/components/video/readme.md]: done in c5c674b 👌

File: src/components/video/video.scss (hunk 7 above)
Comment (MarkMed) [reply src/components/video/video.scss]: feature added in 602551b 👍

**Pull request merged**